### PR TITLE
Update content: `Hds::Table` component

### PR DESCRIPTION
**Pull request by MelSumner**
### :pushpin: Summary

If merged, this PR updates the content for the `table` component.

### CODE REVIEW

File: website/docs/components/table/partials/code/how-to-use.md [hunk 1]
```diff
@@ -25,7 +27,7 @@ If you have your own content and don't want to use a model, you can still benefi
 </Hds::Table>
```
Comment (jorytindall) [inline website/docs/components/table/partials/code/how-to-use.md:27]: Question: for some of these examples do we want to use the code block to show the rendered component? Might be tough in this layout, but just curious

File: website/docs/components/table/partials/guidelines/guidelines.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    <:head as |H|>
+        <H.Tr>
+            <H.Th>Start</H.Th>
+            <H.Th>Middle</H.Th>
+            <H.Th>End</H.Th>
+        </H.Tr>
+    </:head>
+</Hds::Table>
+
+- Column placement determines the visual styling based on where the column is placed relatively to other columns in a table header.
```
Comment (jorytindall) [inline website/docs/components/table/partials/guidelines/guidelines.md:25]:
```suggestion
Column placement determines the visual styling based on where the column is placed relatively to other columns in a table header.
```

File: website/docs/components/table/partials/guidelines/guidelines.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+- Sorting is not relevant for all content, therefore the default sort variant of the header column is none.
+  - Whether a column is sortable is left up to the consumer building the table and should be addressed on a case-by-case basis.
+- Columns that do not contain a sortable data type are interactive and therefore have corresponding hover, active, and focus states.
+- A table may only be sorted by a single value at a time.
+
+### State
+
+![Header column state example](/assets/components/table/table-header_column-state.png)
+
+_Banner (informational):_ Only header columns that contain a sortable data type have state variants. Non-sortable header columns are not interactive and therefore do not have interactive states.
```
Comment (jorytindall) [inline website/docs/components/table/partials/guidelines/guidelines.md:64]: !!! Info

Only header columns that contain a sortable data type have state variants. Non-sortable header columns are not interactive and therefore do not have interactive states.
!!!
```suggestion
!!! Info

Only header columns that contain a sortable data type have state variants. Non-sortable header columns are not interactive and therefore do not have interactive states.
!!!
```

File: website/docs/components/table/partials/specifications/anatomy.md [hunk 4]
```diff
@@ -0,0 +1,11 @@
+## Anatomy
```
Comment (jorytindall) [inline website/docs/components/table/partials/specifications/anatomy.md:1]: This probably also needs the cell anatomy and to be organized into a table, but can be added in a follow-up PR

**Review by jorytindall — APPROVED** [review]
Left a couple of comments, but overall LGTM 🚀

File: website/docs/components/table/partials/guidelines/guidelines.md [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+            <H.Th>Options</H.Th>
+        </H.Tr>
+    </:head>
+</Hds::Table>
+
+### Column width
+
+Within Figma column width is determined manually by resizing the header column and cells. As a best practice, column width should be adjusted to fit the longest data type within the cell.
+
+### Content
```
Comment (Dhaulagiri) [inline website/docs/components/table/partials/guidelines/guidelines.md:87]:
```suggestion
## Content
```

File: website/docs/components/table/partials/guidelines/guidelines.md [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+
+#### Label
```
Comment (Dhaulagiri) [inline website/docs/components/table/partials/guidelines/guidelines.md:89]:
```suggestion
### Label
```

File: website/docs/components/table/partials/specifications/anatomy.md [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+
+![Table header anatomy](/assets/components/table/table-header_column-anatomy.png)
+
+### Label
```
Comment (Dhaulagiri) [inline website/docs/components/table/partials/specifications/anatomy.md:5]: As mentioned above, this should be moved to a table similar to the alerts page

File: website/docs/components/table/partials/guidelines/guidelines.md [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+![Header column sorting](/assets/components/table/table-header_column-sorting.png)
+
+#### Sorting Best practices
+
+- Sorting is not relevant for all content, therefore the default sort variant of the header column is none.
+  - Whether a column is sortable is left up to the consumer building the table and should be addressed on a case-by-case basis.
+- Columns that do not contain a sortable data type are interactive and therefore have corresponding hover, active, and focus states.
+- A table may only be sorted by a single value at a time.
+
+### State
```
Comment (Dhaulagiri) [inline website/docs/components/table/partials/guidelines/guidelines.md:60]: this should be in states.md

File: website/docs/components/table/partials/guidelines/guidelines.md [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+- Columns in the end position frequently use end-alignment when displaying non-string/text content.
+
+### Sorting
+
+![Header column sorting](/assets/components/table/table-header_column-sorting.png)
+
+#### Sorting Best practices
+
+- Sorting is not relevant for all content, therefore the default sort variant of the header column is none.
+  - Whether a column is sortable is left up to the consumer building the table and should be addressed on a case-by-case basis.
```
Comment (Dhaulagiri) [inline website/docs/components/table/partials/guidelines/guidelines.md:56]:
```suggestion
    - Whether a column is sortable is left up to the consumer building the table and should be addressed on a case-by-case basis.
```

Comment (Dhaulagiri) [reply website/docs/components/table/partials/guidelines/guidelines.md:56]: Although I think this entire line may be able to be removed as the previous line mentions that no sorting is on by default

File: website/docs/components/table/partials/guidelines/guidelines.md [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+            <H.Th>Status</H.Th>
+            <H.Th>Created by</H.Th>
+            <H.Th>Options</H.Th>
+        </H.Tr>
+    </:head>
+</Hds::Table>
+
+### Column width
+
+Within Figma column width is determined manually by resizing the header column and cells. As a best practice, column width should be adjusted to fit the longest data type within the cell.
```
Comment (KristinLBradley) [inline website/docs/components/table/partials/guidelines/guidelines.md:88]:
```suggestion
Within Figma, column width is determined manually by resizing the header column and cells. As a best practice, column width should be adjusted to fit the longest data type within the cell.
```

File: website/docs/components/table/partials/code/how-to-use.md (hunk 1 above)
Comment (MelSumner) [reply website/docs/components/table/partials/code/how-to-use.md:27]: we're talking about it 👍

File: website/docs/components/table/partials/specifications/anatomy.md (hunk 4 above)
Comment (MelSumner) [reply website/docs/components/table/partials/specifications/anatomy.md:1]: If you want to export/upload the Figma images, that would be awesome

**Review by KristinLBradley — APPROVED** [review]

**Pull request merged**